spinalcordtoolbox/scripts/sct_compute_snr.py

```diff
@@ -57,19 +54,40 @@ def get_parser():
         help='Binary (or weighted) mask within which SNR will be averaged. Example: dwi_moco_mean_seg.nii.gz',
         metavar=Metavar.file,
         default='')
+    optional.add_argument(
+        '-m-noise',
+        help="Binary (or weighted) mask within which noise will be calculated. Only valid for '-method single'.",
+        metavar=Metavar.file,
+        default='')
     optional.add_argument(
         '-method',
-        help='R|Method to use to compute the SNR:\n'
-             ' diff (default): Substract two volumes (defined by -vol) and estimate noise variance within the ROI (flag -m is required).\n'
-             ' mult: Estimate noise variance over time across volumes specified with -vol.',
-        choices=('diff', 'mult'),
+        help='R|Method to use to compute the SNR (default: diff):\n'
+             "- diff: Substract two volumes (defined by -vol) and estimate noise variance within the ROI "
+             "(flag '-m' is required). Requires a 4D volume.\n"
+             "- mult: Estimate noise variance over time across volumes specified with '-vol'. Requires a 4D volume.\n"
+             "- single: Estimates noise variance in a 5x5 square at the corner of the image, and average the mean "
+             "signal inside the ROI specified by flag '-m'. The variance and mean are corrected for Rayleigh "
+             "distributions. This corresponds to the cases SNRstd and SNRmean in the Dietrich et al. article. Uses a "
+             "3D or a 4D volume. If a 4D volume is input, the volume to compute SNR on is specified by '-vol'.",
+        choices=('diff', 'mult', 'single'),
         default='diff')
     optional.add_argument(
         '-vol',
-        help='Volumes to compute SNR from. Separate with "," (Example: -vol 0,1), or select range '
-             'using ":" (Example: -vol 2:50). By default, all volumes in series are selected.',
+        help="R|Volumes to compute SNR from. Separate with ',' (Example: '-vol 0,1'), or select range "
+             "using ':' (Example: '-vol 2:50'). If this argument is not passed:\n"
+             "  - For '-method mult', all volumes will be used.\n"
+             "  - For '-method diff', the first two volumes will be used.\n"
+             "  - For '-method single', the first volume will be used.",
         metavar=Metavar.str,
         default='')
+    optional.add_argument(
+        '-rayleigh',
+        type=int,
+        help="Correct for Rayleigh distribution. It is recommended to always use this correction for the 'diff' method "
+             "and to use it with the 'single' method in case the noise mask is taken in a region with low SNR (e.g., "
+             "the air). ",
+        default=1,
+        choices=(0, 1))
```

Comment:
I notice that `-rayleigh` is on by default for all methods (even `'mult'`). Could this be detrimental to users who don't check this carefully?

In other words, would it be better for this to be "opt-in" rather than "opt-out"?

Reply:
> In other words, would it be better for this to be "opt-in" rather than "opt-out"?

Rayleigh _should_ be on for diff, and for single (in most cases). And it is not used for mult

Reply:
Ah, thank you for the clarification. In what cases would someone want to set `-rayleigh 0`, then?

Maybe rather than "`to use it with the 'single' method in case the noise mask is taken in a region with low SNR (e.g., the air).`" we could give an example of when to turn this option off instead (since it's already on by default).

Something like "`This should only be turned off in cases where [...]`".

Reply:
yes good point

Reply:
Resolved by 8e10b2d.